code/game/machinery/engi_adv.dm

```diff
@@ -0,0 +1,240 @@
+		icon_state = "nuclearbomb1"
+	update_icon()
+
+/obj/machinery/construction_nuke/proc/set_active()
+	var/area/A = get_area(src)
+	if(safety && !bomb_set)
+		to_chat(usr, "<span class='danger'>Error: The safety is still on.</span>")
+		playsound(src, 'sound/machines/defib_failed.ogg', 75, 1)
+		return
+	if(!A.blob_allowed)
+		to_chat(usr, "<span class='danger'>Error: The device's safety countermeasures flash red: you cannot arm this device outside of the station.</span>")
+		playsound(src, 'sound/machines/defib_failed.ogg', 75, 1)
+		return
+	timing = !timing
+	if(timing)
+		if(cooldown > world.time)
+			to_chat(usr, "<span class='danger'>Error: The device is still resetting from the last activation, it will be ready again in [round((cooldown-world.time)/10)] seconds.</span>")
+			playsound(src, 'sound/machines/defib_failed.ogg', 75, 1)
+			timing = FALSE
+			return
+		bomb_set = TRUE
+		priority_announce("We are detecting a massive spike of radioactive energy from \a [payload] payload originating from [A.map_name]. If this is not a scheduled occurrence, please investigate immediately.","Nanotrasen Nuclear Safety Division", 'sound/misc/airraid.ogg')
+		cooldown = world.time + 1200
+		detonation_timer = world.time + (timer_set * 10)
+		icon_state = "nuclearbombc"
+	else
+		bomb_set = FALSE
+		priority_announce("Radioactive energy levels are normalizing, please submit an incident report as soon as possible.","Central Command Nuclear Safety Division", 'sound/AI/attention.ogg')
+		detonation_timer = null
+		icon_state = "nuclearbomb1"
+		playsound(src, 'sound/machines/terminal_off.ogg', 75, 1)
+	update_icon()
+
+/obj/machinery/construction_nuke/proc/get_time_left()
+	if(timing)
+		. = round(max(0, detonation_timer - world.time) / 10, 1)
+	else
+		. = timer_set
+
+/obj/machinery/construction_nuke/proc/explode()
+	if(safety || !bomb_set)
+		timing = FALSE
+		return
+	exploding = TRUE
+	update_icon()
+	for(var/mob/M in GLOB.player_list)
+		M << 'sound/machines/Alarm.ogg'
+	addtimer(CALLBACK(src, .proc/boom), 100, TIMER_CLIENT_TIME)
+
+/obj/machinery/construction_nuke/proc/boom()
+	var/turf/startpoint = get_turf(src)
+	new /obj/effect/temp_visual/explosion(startpoint)
+	qdel(src)
+	for(var/I in spiral_range_turfs(range, startpoint))
+		var/turf/T = I
+		if(!T)
+			continue
+		if(istype(T, /turf/open/floor))
+			T.ChangeTurf(payload_floor)
+			new /obj/effect/temp_visual/fire(T)
+		else if(istype(T, /turf/closed/wall))
+			T.ChangeTurf(payload_wall)
+			new /obj/effect/temp_visual/fire(T)
+		CHECK_TICK
+	if((payload == "uranium" || payload == "plasma") && (SSshuttle.emergency.mode == SHUTTLE_RECALL || SSshuttle.emergency.mode == SHUTTLE_IDLE || (SSshuttle.emergency.timeLeft(1) > (SSshuttle.emergencyCallTime * 0.4))))
+		SSshuttle.emergencyNoRecall = TRUE
+		SSshuttle.emergency.request(null, set_coefficient = 0.4)
+		priority_announce("Catastrophic event detected: crisis shuttle protocols activated - jamming recall signals across all frequencies")
```

Comment:
Not sure why not. This uses a new var I added to the emergency shuttle to prevent any sneaky recall methods from working.